code/game/objects/items/tajara.dm

```diff
@@ -146,47 +147,23 @@
 	else
 
 		var/adhomian_year = game_year + 1158
-
-		var/current_month = time2text(world.realtime, "Month")
+		var/current_month = text2num(time2text(world.realtime, "MM"))
 		var/current_day = text2num(time2text(world.realtime, "DD"))
 		var/adhomian_day
-		var/adhomian_month
+		var/adhomian_month = src.months[Ceiling(current_month/3)]
 		switch(current_month)
-			if("January", "February", "March")
-				adhomian_month = "Menshe-aysaif"
-				if (current_month == "February")
-					adhomian_day = current_day + 15
-				if (current_month == "March")
-					adhomian_day = current_day + 30
-
-			if("April", "May", "June")
-				adhomian_month = "Sil'nryy-aysaif"
-				if (current_month == "May")
-					adhomian_day = current_day + 15
-				if (current_month == "June")
-					adhomian_day = current_day + 30
-
-			if("July", "August", "September")
-				adhomian_month = "Menshe-rhazzimy"
-				if (current_month == "August")
-					adhomian_day = current_day + 15
-				if (current_month == "September")
-					adhomian_day = current_day + 30
-
-			if("October", "November", "December")
-				adhomian_month = "Sil'nryy-rhazzimy"
-				if (current_month == "November")
-					adhomian_day = current_day + 15
-				if (current_month == "December")
-					adhomian_day = current_day + 30
-
+			if(2, 5, 8, 11)
+				current_day += 31
+			if(6, 9, 12)
+				current_day += 61
+			if(3)
+				current_day += 59 + isLeap(text2num(time2text(world.realtime, "YYYY")))
```

Comment:
As discussed on Discord I'm not really convinced this is explicit enough. It looks like a mistake.

I'd either add a comment like this:
```suggestion
				current_day += 59 + isLeap(text2num(time2text(world.realtime, "YYYY"))) // we can conveniently use the result of `isLeap` to add 1 when we are in a leap year
```

or used a ternary to be more explicit:

```suggestion
				current_day += 59 + ( isLeap(text2num(time2text(world.realtime, "YYYY"))) ? 1 : 0)
```

or I'd separate it into branches:

```suggestion
				if(isLeap(text2num(time2text(world.realtime, "YYYY"))))
					current_day += 60
				else
					current_day += 59
```

Pick your poison 🙂.